doc/user/content/transform-data/patterns/percentiles.md

```diff
@@ -10,134 +10,306 @@ menu:
+The following example assumes you have not previously created and populated the
+`input` table from the [Using histograms to compute percentiles
+example](#example). If you have created and populated the table, skip to step 3.
 
-HDR histograms are related to how [floating point numbers are represented](https://en.wikipedia.org/wiki/Double-precision_floating-point_format) as integers. The underlying assumption is that smaller numbers require a higher precision to be distinguishable (e.g. 5 ms and 6 ms are different and should be in different buckets) whereas larger numbers can be rounded more aggressively as their relative error becomes less relevant (e.g. 10000 ms and 10001 ms are almost the same and can reside in the same bucket).
+{{</ tip >}}
 
-The following snippets reduce the precision of numbers to limit the amount of required buckets. Numbers are decomposed into their floating point representation
+1. Create a table `input`:
 
-	n = sign * mantissa * 2**exponent
+   ```mzsql
+   CREATE TABLE input (value BIGINT);
+   ```
 
-and the precision of the mantissa is then reduced to compute the respective bucket.
+2. Insert into the `input` table values `1` to `10`.
 
-The basic ideas of using histograms to compute percentiles remain the same; but determining the bucket becomes more involved because it’s now composed of the triple (sign, mantissa, exponent).
+   ```mzsql
+   INSERT INTO input
+   SELECT n FROM generate_series(1,10) AS n;
+   ```
 
-```mzsql
--- precision for the representation of the mantissa in bits
-\set precision 4
+1. Create a `hdr_histogram` view. To reduce the number of buckets, the values
+   are first decomposed into their floating point representation (i.e.,
+   `mantissa * 2^exponent`), and with the precision of the mantissa is reduced
+   to 4, reconstructed.
 
-CREATE VIEW hdr_histogram AS
-SELECT
-  CASE WHEN value<0 THEN -1 ELSE 1 END AS sign,
-  trunc(log(2.0, abs(value)))::int AS exponent,
-  trunc(pow(2.0, :precision) * (value / pow(2.0, trunc(log(2.0, abs(value)))::int) - 1.0))::int AS mantissa,
-  count(*) AS frequency
-FROM input
-GROUP BY sign, exponent, mantissa;
-```
+   {{< tabs >}}
 
-The `hdr_distribution` view below reconstructs the `bucket` (with reduced precision), and determines the cumulative count and cumulative distribution.
+   {{< tab "Materialize Console" >}}
 
 ```mzsql
-CREATE VIEW hdr_distribution AS
+CREATE VIEW hdr_histogram AS
+WITH
+  input_parts AS (
+    SELECT
+      CASE WHEN value = 0 THEN NULL
+          ELSE trunc(log(2, abs(value)))::int
+      END AS exponent,
+      CASE WHEN value = 0 THEN NULL
+          ELSE value / pow(2.0, trunc(log(2, abs(value)))::int)
+      END AS mantissa
```

Comment:
I understand the spirit of what is going on here, but perhaps a good moment to put a comment in describing the properties of this field? Something like `-- A signed value in [-1, -0.5) or (0.5, 1] that multiplies 2^exponent to reproduce value`

Reply:
This part is made awkward by SQL limitations. Ideally this would be `value / pow(2.0, exponent)`, which would be quite obvious imo (especially because we point out the `mantissa * 2^exponent` representation above. But we can't name `exponent` here :/